barretenberg/cpp/src/barretenberg/polynomials/polynomial.cpp

```diff
@@ -267,6 +268,36 @@ Fr Polynomial<Fr>::evaluate_from_fft(const EvaluationDomain<Fr>& large_domain,
     return polynomial_arithmetic::evaluate_from_fft(coefficients_.get(), large_domain, z, small_domain);
 }
 
+template <typename Fr> void Polynomial<Fr>::set_to_right_shifted(std::span<Fr> coeffs_in, size_t shift_size)
+{
+    // Ensure we're not trying to shift self
+    ASSERT(coefficients_.get() != coeffs_in.data());
+
+    auto size_in = coeffs_in.size();
+    ASSERT(size_in > 0);
+    ASSERT(shift_size < size_in);
+
```

Comment:
Why are you comparing shift size with input size?

Reply:
Basically just because I'm about to loop through shift-size many coefficients of the input. (Changed to `shift_size <= size_in` accordingly). Added a TODO here with reference to existing bberg issue #723 related to allowing the right-shifted transcript polynomials to have size greater than `n`.